src/win/tcp.c

```diff
@@ -999,7 +1000,8 @@ void uv_process_tcp_read_req(uv_loop_t* loop, uv_tcp_t* handle,
     }
 
     /* Do nonblocking reads until the buffer is empty */
-    while (handle->flags & UV_HANDLE_READING) {
+    count = 32;
+    while (handle->flags & UV_HANDLE_READING && count-- > 0) {
```

Comment:
Can you put parens around the `handle->flags & UV_HANDLE_READING`? mingw-gcc will complain otherwise, I'm reasonably sure.